klpbuild/config.py

```diff
@@ -61,20 +68,54 @@ def __init__(self, lp_name, lp_filter, kdir=False, data_dir=None, skips="", work
         self.kdir = self.conf.get("kdir", False)
         self.data = Path(self.conf.get("data", "non-existent"))
         if not self.data.exists():
-            data_dir = os.getenv("KLP_DATA_DIR", "")
-            if not data_dir:
-                raise ValueError("KLP_DATA_DIR should be defined")
-            self.data = Path(data_dir)
-
-        if not self.data.is_dir():
-            raise ValueError("Data dir should be a directory")
+            self.data = self.get_user_path('data_dir')
 
         # will contain the symbols from the to be livepatched object
         # cached by the codestream : object
         self.obj_symbols = {}
 
         logging.basicConfig(level=logging.INFO, format="%(message)s")
 
+    def setup_user_env(self, basedir):
+        basedir = Path(basedir, "klp-build")
+        workdir = Path(basedir, "patches")
```

Comment:
I would prefer having the directory "livepatches" over patches, since it's common to have a "patches" directory. Stating as "livepatches" directory seems more objective.